File: blog-server/src/module/admin/links/Links.service.ts
import {Injectable, NotFoundException} from '@nestjs/common'
import { InjectRepository } from '@nestjs/typeorm'
import { Repository } from 'typeorm'
import { Links } from '../../../db/entities/Links'
import {LinksInsertParam, LinksUpdateParam} from "./model/links.param";
import {PageRsult} from "../../../model/BaseRsult";
import {IdParam, PageParam} from "../../../model/BaseParam";

@Injectable()
export class LinksService {
    constructor(
        @InjectRepository(Links) private linksRepository: Repository<Links>,
    ) {}

    // 新增
    async insert(body: LinksInsertParam): Promise<void> {
        const link = new Links()
        link.name = body.name
        link.avatar = body.avatar
        link.http_url = body.http_url
        link.status = body.status
        link.description = body.description
        await this.linksRepository.save(link)
    }


    // 修改
    async update(body: LinksUpdateParam): Promise<void> {
        const link = await this.linksRepository.findOne({where: { id: body.id }})
        if (!link) throw new NotFoundException("不存在该项目")
        link.name = body.name
        link.avatar = body.avatar
        link.http_url = body.http_url
        link.status = body.status
        link.description = body.description
        await this.linksRepository.save(body)
    }

    // 查询
    async list(body: PageParam): Promise<PageRsult<Links>> {
        const list = await this.linksRepository.find({
            skip: body.pageNum * ( body.page - 1),
            take: body.pageNum
        })
        const total = await this.linksRepository.count()
        return {
            list,
            total
        }
    }


    // 删除
    async delete(body: IdParam) :Promise<void> {
        await this.linksRepository.delete({ id: body.id })
    }

}

File: BlogServer/Blog.Service/Api/LinksService.cs
using Blog.Model.Entity;
using Blog.Model.Param;
using Blog.Model.Rsult;
using Blog.Service.Api.Interface;
using Blog.Utils;
using SqlSugar;
using System.Xml.Linq;

namespace Blog.Service.Api
{
    internal class LinksService: ILinksService
    {
        public LinksService(ISqlSugarClient client)
        {
            Db = client;
        }

        public ISqlSugarClient Db { get; }

        public async Task Insert(LinksInsertParam param)
        {
            //var element = new LinksEnity
            //{
            //    Name = param.Name,
            //    Avatar = param.Avatar,
            //    Http_url = param.Http_url,
            //    Status = param.Status,
            //    Description = param.Description
            //};
            var element = new LinksEnity();
            CommonFun.AssignProps(element, param);
            await Db.Storageable(element).ExecuteCommandAsync();
        }

        public async Task Update(LinksUpdateParam param)
        {
            var link = await Db.Queryable<LinksEnity>().Where(it => it.Id == param.Id).FirstAsync();
            //link.Name = param.Name;
            //link.Avatar = param.Avatar;
            //link.Http_url = param.Http_url;
            //link.Status = param.Status;
            //link.Description = param.Description;
            CommonFun.AssignProps(link, param);
            await Db.Storageable(link).ExecuteCommandAsync();
        }

        public async Task<PageResult<LinksEnity>> List(PageParam param)
        {
            RefAsync<int> total = 0;
            var list = await Db.Queryable<LinksEnity>().ToPageListAsync(param.Page, param.PageNum, total);
            return new PageResult<LinksEnity>
            {
                Total = total,
                List = list
            };
        }

        public async Task Delete(IDParam param)
        {
            await Db.Deleteable<LinksEnity>().In(it => it.Id, param.Id).ExecuteCommandAsync();
        }
    }
}
